tests/templates/test_templates.py

```diff
@@ -550,6 +550,55 @@ def test_get_type_by_name(self, client, mock_template):
         new_type = client.get_templatetype_by_name(type.template_id, type.name)
         assert new_type is not None, "Resource type attrs not retrived by name!"
 
+    def test_clone_templatetype(self, client, mock_template):
+        """Test cloning a template type"""
+        original_template = client.get_template(mock_template.id)
+        original_type = original_template.templatetypes[0]
+        
+        # Clone the template type
+        cloned_type = client.clone_templatetype(original_type.id)
+        cloned_type_j = JSONObject(cloned_type)
+        
+        # Verify the clone was created successfully
+        assert cloned_type_j is not None, "Cloned template type should not be None"
+        assert cloned_type_j.id is not None, "Cloned template type should have an ID"
+        assert cloned_type_j.id != original_type.id, "Cloned type should have different ID than original"
+        
+        # Verify that only the specified fields are different
+        # Name should be modified to be unique (appends "(Clone)")
+        assert cloned_type_j.name == original_type.name + " (Clone)", "Cloned type should have modified name"
```

Comment:
A minor observation: the checked-for name has the format "xxx (Clone)" only in the test where there is guaranteed to be no name collision that required adding some digits in `clone_templatetype()`.

Otherwise, the returned name will have some pattern including "(Clone)" and potentially digits at the end.